GHC 9.4: warning about missing `LANGUAGE TypeOperators`

This warning can be fixed in a backwards-compatible way by adding `TypeOperators` to `default-extensions:` in the `.cabal` file.
```
Building library for dlist-1.0..
[1 of 5] Compiling Data.DList.Internal ( Data/DList/Internal.hs, dist/build/Data/DList/Internal.o, dist/build/Data/DList/Internal.dyn_o )

Data/DList/Internal.hs:642:12: warning: [-Wtype-equality-requires-operators]
    The use of ‘~’ without TypeOperators
    will become an error in a future GHC release.
    Suggested fix: Perhaps you intended to use TypeOperators
    |
642 | instance a ~ Char => IsString (DList a) where
    |            ^
[2 of 5] Compiling Data.DList       ( Data/DList.hs, dist/build/Data/DList.o, dist/build/Data/DList.dyn_o )
[3 of 5] Compiling Data.DList.DNonEmpty.Internal ( Data/DList/DNonEmpty/Internal.hs, dist/build/Data/DList/DNonEmpty/Internal.o, dist/build/Data/DList/DNonEmpty/Internal.dyn_o )

Data/DList/DNonEmpty/Internal.hs:459:12: warning: [-Wtype-equality-requires-operators]
    The use of ‘~’ without TypeOperators
    will become an error in a future GHC release.
    Suggested fix: Perhaps you intended to use TypeOperators
    |
459 | instance a ~ Char => IsString (DNonEmpty a) where
    |            ^
```

--- a/dlist.cabal
+++ b/dlist.cabal
@@ -54,6 +54,7 @@ library
     exposed-modules:    Data.DList.DNonEmpty
     other-modules:      Data.DList.DNonEmpty.Internal
   default-language:     Haskell2010
+  default-extensions:   TypeOperators
   ghc-options:          -Wall
   if impl(ghc >= 8.0)
     ghc-options:        -Wcompat

